### ISSUE TITLE
certbot adds acme-challenge to wrong config, if domain name is not lower case

### ISSUE THREAD

**Issue by SlavikCA** [certbot/certbot#6776]
- Debian 9.6
- I installed Certbot with OS package manager
- certbot 0.28.0
- nginx

- had this host in my config in `/etc/nginx/conf.d/RussianChurchVancouver.ca.conf`:  
```
server {
  listen       80;
  server_name  RussianChurchVancouver.ca www.RussianChurchVancouver.ca;
...
```

- I ran certbot and certs were not created, because certbot failed:
```
root@deb96:/var/www/RussianChurchVancouver.ca/web# certbot -m slavikca@gmail.com --agree-tos --nginx -d RussianChurchVancouver.ca -d www.RussianChurchVancouver.ca
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator nginx, Installer nginx
Obtaining a new certificate
Performing the following challenges:
http-01 challenge for russianchurchvancouver.ca
http-01 challenge for www.russianchurchvancouver.ca
nginx: [warn] conflicting server name "www.russianchurchvancouver.ca" on 0.0.0.0:80, ignored
nginx: [warn] conflicting server name "russianchurchvancouver.ca" on 0.0.0.0:80, ignored
Waiting for verification...
Cleaning up challenges
Failed authorization procedure. russianchurchvancouver.ca (http-01): urn:ietf:params:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from http://russianchurchvancouver.ca/.well-known/acme-challenge/b4vlj_zIdB4H_a94FlltJWx2JBGBDS_ihAFWrpJSl5U: "<html>\r\n<head><title>404 Not Found</title></head>\r\n<body>\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx</center>\r\n", www.russianchurchvancouver.ca (http-01): urn:ietf:params:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from http://www.russianchurchvancouver.ca/.well-known/acme-challenge/B8e0i1p0jhjNtfV1Dd36rlE8eh4K2bpefVvABc6Na48: "<html>\r\n<head><title>404 Not Found</title></head>\r\n<body>\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx</center>\r\n"
```

While troubleshooting, I found, that  certbot adding acme-challenge to this config file:
```
Writing nginx conf tree to /etc/nginx/conf.d/default.conf
```
which is the bug. that's why I have `conflicting server name` in the log (certbot created situation when I have `russianchurchvancouver.ca` both in `default.conf` and in `RussianChurchVancouver.ca.conf`
Expected, that it would add acme-challenge to this file:
```
/etc/nginx/conf.d/RussianChurchVancouver.ca.conf
```

At this point, I started to suspect, that certbot does that because my domains in config file are not lower case.
So, I lowercased them:
```
  server_name  russianchurchvancouver.ca www.russianchurchvancouver.ca;
```
and certbot worked correctly:
```
root@deb96:/etc/nginx/conf.d# certbot -m slavikca@gmail.com --agree-tos --nginx -d russianchurchvancouver.ca,www.russianchurchvancouver.ca
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator nginx, Installer nginx
Obtaining a new certificate
Performing the following challenges:
http-01 challenge for www.russianchurchvancouver.ca
http-01 challenge for russianchurchvancouver.ca
Waiting for verification...
Cleaning up challenges
Deploying Certificate to VirtualHost /etc/nginx/conf.d/RussianChurchVancouver.ca.conf
Deploying Certificate to VirtualHost /etc/nginx/conf.d/RussianChurchVancouver.ca.conf
...
Congratulations! You have successfully enabled https://russianchurchvancouver.ca
and https://www.russianchurchvancouver.ca
```

**Comment by ohemorange** [comment]
Thanks for reporting, this sure is a bug! You're right, the fix for now is to make sure to match the case to what's written in the config files.

**Comment by ohemorange** [comment]
This is still a bug.

**Comment by alexanderjanusz** [comment]
This is still an issue, and changing the config file to lowercase still fixes it. Calling certbot with matching/uppercase does not work.

**Comment by osirisinferi** [comment]
@alexanderjanusz I can't reproduce this issue any longer locally with a configuration file called `Example.Com` and containing `server_name Example.Com`. Certbot recognises this configuration perfectly and provides the lowercase hostname `example.com`.

Which Certbot version are you running and could you provide the full log file of a Certbot run with the error?

**Comment by alexanderjanusz** [comment]
Hi, 

 

I installed Certbot following the digital ocean tutorial

 

https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-20-04

 

I suspect they may have an outdated certbot version/modified code, now that you highlight the discrepancy. My console tells me I’m using version “certbot 0.40.0”

 

Best regards,

Alex

**Comment by alexanderjanusz** [comment]
> @alexanderjanusz I can't reproduce this issue any longer locally with a configuration file called `Example.Com` and containing `server_name Example.Com`. Certbot recognises this configuration perfectly and provides the lowercase hostname `example.com`.
> 
> Which Certbot version are you running and could you provide the full log file of a Certbot run with the error?

I installed Certbot following the digital ocean tutorial

https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-20-04

I suspect they may have an outdated certbot version/modified code, now that you highlight the discrepancy. My console tells me I’m using version “certbot 0.40.0”

**Comment by osirisinferi** [comment]
> I suspect they may have an outdated certbot version/modified code, now that you highlight the discrepancy. My console tells me I’m using version “certbot 0.40.0”

It is to be expected you're still running into this issue. Version 0.40.0 did not contain the fix for this issue, as this issue was fixed in 1.19.0.
Packages of official OS repositories are often quite outdated, which is the case for Ubuntu. Please see the recommended installation guide at https://certbot.eff.org/
